gax/call_context.h

```diff
@@ -157,9 +177,31 @@ class CallContext {
    */
   MethodInfo Info() const;
 
+  /**
+   * Setter for call-specific retry policy.
+   */
+  void SetRetryPolicy(gax::RetryPolicy const& retry_policy);
+
+  /**
+   * Getter for retry policy.
+   */
+  std::unique_ptr<gax::RetryPolicy> RetryPolicy() const;
+
+  /**
+   * Getter for backoff policy.
+   */
+  std::unique_ptr<gax::BackoffPolicy> BackoffPolicy() const;
+
+  /**
+   * Setter for call-specific backoff policy.
+   */
+  void SetBackoffPolicy(gax::BackoffPolicy const& backoff_policy);
+
  private:
   FRIEND_TEST(CallContext, Basic);
```

Comment:
I _think_ I added this instance of FRIEND_TEST into the code before we discussed that. I'm just about to do a big cleanup review, and I'll try to change the test in that.

Reply:
In this specific case, the _correct_ way to test would be to set some metadata, prepare a `grpc::ClientContext`, and check that it had the correct metadata set. The problem with that is that ClientContext's metadata attribute is private with a *lot* of friends, instead of a const accessor like God intended.

I can absolutely add an accessor here for `gax::CallContext`'s metadata, but the test still isn't really comprehensive for reasons beyond our control.